File: application/src/main/java/com/study/platform/domain/team/application/FindTeamMembersService.java
package com.study.platform.domain.team.application;

import com.study.platform.domain.team.dto.response.TeamMemberResponse;
import com.study.platform.domain.team.port.TeamMemberQueryPort;
import com.study.platform.domain.team.usecase.FindTeamMembersUseCase;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.UUID;

@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class FindTeamMembersService implements FindTeamMembersUseCase {

    private final TeamMemberQueryPort teamMemberQueryPort;

    @Override
    public List<TeamMemberResponse> execute(UUID teamId) {
        return teamMemberQueryPort.findAllByTeamId(teamId);
    }
}

File: application/src/main/kotlin/com/study/platform/domain/team/application/FindTeamMembersService.kt
package com.study.platform.domain.team.application

import com.study.platform.domain.team.dto.response.TeamMemberResponse
import com.study.platform.domain.team.model.TeamMemberRepository
import com.study.platform.domain.team.port.TeamMemberQueryPort
import com.study.platform.domain.team.usecase.FindTeamMembersUseCase
import com.study.platform.global.exception.CustomException
import com.study.platform.global.exception.ErrorCode
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.util.UUID

@Service
@Transactional(readOnly = true)
class FindTeamMembersService(
    private val teamMemberQueryPort: TeamMemberQueryPort,
    private val teamMemberRepository: TeamMemberRepository
) : FindTeamMembersUseCase {

    override fun execute(userId: UUID, teamId: UUID): List<TeamMemberResponse> {
        if (!teamMemberRepository.existsByTeamIdAndUserId(teamId, userId)) {
            throw CustomException(ErrorCode.NOT_TEAM_MEMBER)
        }
        return teamMemberQueryPort.findAllByTeamId(teamId)
    }
}
